### ISSUE TITLE
Unable to update hotspot location

### ISSUE THREAD

**Issue by Debelizmaj81** [helium/wallet-app#683]
**Describe the bug**
On 10.03.2024. at 12:09 I updated location of a Bobcat miner (transaction signature: 5BVvK4kbQQHd9zfGjWz5E9zj6MHewP645LTsjF6METFwBfaPLACqXA1y3Mofnk5UC88tRjnsV36tPcBX5W3bFSMj) that I transferred from Montenegro to Central Serbia. In helium wallet everything went ok, approx. 10 usd fee was charged in HNT, and in notifications it said "App Interaction - Success", but the location hasn't changed. Meanwhile I changed location of one Panther X2 hotspot and it said success and the location was updated in a few minutes. I again tried to reassert location of Bobcat a few hours later (transaction signature: E1be9RGJ9NZe1XG3V1PPZov91oJsEJYHh2kDgy28UtXXjFvXL8HaBo8UBjuMBSoZk1tHgEeRxdtd85k6rSXwed2), and again it was successful in the helium wallet but location did not change and again I kept getting invalid witnesses and beacons due to max distance exceeded. Any solution?

**To Reproduce**

**Expected behavior**

**Screenshots**

**Smartphone (please complete the following information):**

- App Version [e.g. 3.3.0]

**Additional context**
Previous physical location of the hotspot (and still valid on the explorer) is [42.44842637701735, 18.646822252304535], and the real location of the hotspot is [43.864734, 20.372734]

**Comment by ChewingGlass** [comment]
Where are you checking the location of the hotspot? Due to caching it may take up to a day for changes to show?

**Comment by Debelizmaj81** [comment]
Both in Moken explorer and in HeliumGeek it says the same location (the old one in Montenegro). And all of the activity is invalid due to maximum distance exceeded. I turned the device off in order to avoid being denylisted. Until the problem is solved. The name of the hotspot is Gigantic Seafoam Buffalo. I mean I can keep trying to update location but how can I know that I won't be charged another 10 usd without success, like previous two times.

**Comment by ChewingGlass** [comment]
Everything is showing matching on chain and on our api (https://entities.nft.helium.io/11PX1Uom3dw27657RWQNWyb23biDXfxF6PcX8FMJKKu2yRHY7ab)

2 days ago you ran an update but did not change the location. Not sure if that helps.

**Comment by ChewingGlass** [comment]
It does not look as though you were ever charged $10 USD on this hotspot.

**Comment by Debelizmaj81** [comment]
I am sure that I've been charged twice for location upgrade just for this hotspot. Is there any way for me to prove this? Some balance tracking in the wallet? It is a serious bug if I was charged and there is no trace of transaction. I am a layman in coding but there is a line in the final section in the transaction details that goes like this: 
> Program logged: "HNT Price is 924980356 * 10^-8, issuing 1000000 data credits"
If I perform location update again can I make a video od the whole process just to have evidence?

**Comment by Debelizmaj81** [comment]
I just checked in my wallet and I have 1.000.000 DC. It didn't occur to me that this could happen. So I successfully updated the location (according to the app) but the location was not updated but instead I got 1.000.000 DC? Also for the second transaction there is a similar line in the code in the transaction details:

> Program logged: "HNT Price is 923017514 * 10^-8, issuing 1000000 data credits"

But the location was still not updated and I don't have 2.000.000 data credits in the wallet, just 1.000.000.

And just to be clear, it's not about money, it's about solving a problem. I invested a lot of money in 11 hotspot at peak prices, so it's not that I care about 10 usd. I just want to enable the hotspot.
